patches/tModLoader/Terraria/Projectile.cs.patch

```diff
@@ -162,14 +162,14 @@
  				if (ProjectileID.Sets.NeedsUUID[Type])
  					projectile.projUUID = projectile.identity;
  
-@@ -8042,6 +_,8 @@
- 				}
- 			}
+@@ -8158,6 +_,8 @@
+ 			if (Type == 249)
+ 				projectile.frame = Main.rand.Next(5);
  
 +			ProjectileLoader.OnSpawn(projectile, spawnSource);
```

Comment:
Why move this to the end of the method? Having it before the `NetMessage.SendData` call would be better as modders wouldn't have to wait an update tick for their changes to sync. If the issue is that the code section after the `NetMessage.SendData` call would override whatever changes were made in `OnSpawn`, that section could be made to run before it using gotos.

Reply:
Indeed, I think the easiest solution is to move the `NetMessage.SendData` block to the end of the method, which I did.